shared/src/main/scala/mlscript/ConstraintSolver.scala

```diff
@@ -271,7 +271,7 @@ class ConstraintSolver extends NormalForms { self: Typer =>
         }
       }
     
-    def rec(lhs: SimpleType, rhs: SimpleType, sameLevel: Bool, nestedProv: Opt[TypeProvenance] = N)
+    def rec(lhs: SimpleType, rhs: SimpleType, sameLevel: Bool, nestedProv: Opt[NestedTypeProvenance] = N)
```

Comment:
It seems the `sameLevel` boolean has become redundant. Eventually it would have the same info as `nestedProv.isEmpty`, rigth?